### DISCUSSION TITLE
can you directly delete files from storage via supabase dashboard/ui?

### DISCUSSION THREAD

**Discussion by devhandler** [supabase/supabase#16393] [Questions] [answered]
or will that only remove the metadata in postgres storage.objects table but leave the actual files orphaned in the bucket?

**Comment by GaryAustin1** [comment, marked as answer]
If you mean using the table UI or SQL editor directly on storage.objects, No.  You have to make storage-api calls.  If you delete just storage.objects row the file data is orphaned and no way to get rid of it.

Here is my approach for managing it.   https://github.com/GaryAustin1/supa-file-helper
Which is overkill for a single file.

The storage UI in the dashboard will delete the s3 data as well as the metadata.

**Reply by devhandler** [reply]
thank you! yep,i meant the storage ui. sorry for the confusion. glad to hear it delete actual files.

**Comment by MadeByDouglas** [comment]
so wait, then what the people say on this thread https://github.com/orgs/supabase/discussions/3124 is not true then? Doing an SQL query just orphans the data and doesn't actually delete it?

**Reply by GaryAustin1** [reply]
I unselected the old answer on that thread although there are lots of corrections later that you must call the storage api to delete the file.

**Reply by MadeByDouglas** [reply]
ok I see, thank you, I understand now, it was confusing since it was marked as "the answer"
